docker/supabase/volumes/db/init/04-telescope-schema.sql

```diff
@@ -0,0 +1,41 @@
+-- Case-insensitive text type
+CREATE EXTENSION IF NOT EXISTS citext WITH SCHEMA extensions;
+
+-- Registered Telescope users
+CREATE TABLE IF NOT EXISTS telescope_profiles (
+  id text PRIMARY KEY,
+  registered_at timestamptz DEFAULT timezone('utc'::text, now()) NOT NULL,
+  updated_at timestamptz DEFAULT timezone('utc'::text, now()) NOT NULL,
+  display_name citext UNIQUE NOT NULL,
+  first_name text NOT NULL,
+  last_name text NOT NULL,
+  email citext UNIQUE NOT NULL,
+  github_username citext UNIQUE NOT NULL,
+  github_avatar_url text NOT NULL
```

Comment:
Would we want to bake in the GitHub information inside the Telescope profile?

Reply:
We assume all Telescope users would have a GitHub profile. Thinking aloud. An alternative approach could be having a separate table for GitHub information and forming a relationship between the two tables.

```
CREATE TABLE IF NOT EXISTS telescope_profiles (
  id text PRIMARY KEY,
  registered_at timestamptz DEFAULT timezone('utc'::text, now()) NOT NULL,
  updated_at timestamptz DEFAULT timezone('utc'::text, now()) NOT NULL,
  display_name citext UNIQUE NOT NULL,
  first_name text NOT NULL,
  last_name text NOT NULL,
  email citext UNIQUE NOT NULL
);

CREATE TABLE IF NOT EXISTS github_info (
  github_username citext UNIQUE NOT NULL,
  github_avatar_url text NOT NULL,
  CONSTRAINT fk_telescope_profiles FOREIGN KEY(github_username) REFERENCES telescope_profiles(id)
);
```
This way when a user signs up on Telescope, they still have to provide a GitHub account but the GitHub data is kept in another table.

I believe GitHub usernames can be changed so we probably need to use something else the primary key.

Untested so this is just a thought.

Reply:
I'm not sure what you mean here.

Reply:
I don't see any benefits of managing an extra table to store GitHub data.

Having another table we would have to do more queries to register a user, get user info, and manage additional RLS policies for that table.

Reply:
Right now, we're only using the github username and avatar URL, but what if we want to use more github user info in the future? Say we want to have a User Profile page on Telescope and want to display a few more interesting data points from the user's github like how many followers they have, if they're hireable, how long they've been on GitHub, etc.

Reply:
I don't think it's a good idea to manage that, have to keep it in sync with GitHub, etc. Why don't we use GitHub APIs to get more user info if we ever want to?

Reply:
I just read the GitHub Docs, you're right, it would make sense to just use the GitHub API in this case.